datamodel/core.py

```diff
@@ -0,0 +1,373 @@
+            raise TypeError("Datamodel %r must have a _name" % cls)
+
+        # all datamodels except 'base' implicitly inherit from 'base'
+        if name != "base":
+            parents = list(parents) + ["base"]
+
+        # create or retrieve the datamodel's class
+        if name in parents:
+            if name not in registry:
+                raise TypeError(
+                    "Datamodel %r does not exist in registry." % name
+                )
+
+        # determine all the classes the datamodel should inherit from
+        bases = LastOrderedSet([cls])
+        for parent in parents:
+            if parent not in registry:
+                raise TypeError(
+                    "Datamodel %r inherits from non-existing datamodel %r."
+                    % (name, parent)
+                )
+            parent_class = registry[parent]
+            if parent == name:
+                for base in parent_class.__bases__:
+                    bases.add(base)
+            else:
+                bases.add(parent_class)
+                parent_class._inherit_children.add(name)
+
+        if name in parents:
+            DatamodelClass = registry[name]
+            # Add the new bases to the existing model since the class into
+            # the registry is could already be used into an inherit
+            DatamodelClass.__bases__ = tuple(bases)
+            # We must update the marshmallow schema on the existing datamodel
+            # class to include those inherited
+            parent_schemas = []
+            for parent in bases:
+                if issubclass(parent, MarshmallowModel):
+                    parent_schemas.append(parent.__schema_class__)
+            schema_class = type(name + "Schema", tuple(parent_schemas), {})
+            DatamodelClass.__schema_class__ = schema_class
+        else:
+            attrs = {
+                "_name": name,
+                "_register": False,
+                # names of children datamodel
+                "_inherit_children": OrderedSet(),
+            }
+            if name == "base":
+                attrs["_registry"] = registry
+            DatamodelClass = type(name, tuple(bases), attrs)
+
+        setattr(DatamodelClass.__schema_class__, "_registry", registry)
+        setattr(DatamodelClass.__schema_class__, "_datamodel_name", name)
+        setattr(
+            DatamodelClass.__schema_class__, "__make_object__", __make_object__
+        )
+        DatamodelClass._complete_datamodel_build()
+
+        registry[name] = DatamodelClass
+
+        return DatamodelClass
+
+    @classmethod
+    def _complete_datamodel_build(cls):
+        """ Complete build of the new datamodel class
+
+        After the datamodel has been built from its bases, this method is
+        called, and can be used to customize the class before it can be used.
+
+        Nothing is done in the base Datamodel, but a Datamodel can inherit
+        the method to add its own behavior.
+        """
+
```

Comment:
```suggestion
pass
```
To avoid "indent expected" error